### PR TITLE
fix: added utility functions in order to encode textual features into numerical features 

### PR DESCRIPTION
**Pull request by aiday-mar**

### CODE REVIEW

File: src/utils/numericEncodingOfArray.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+/**
+ * Numerically encodes the strings in the array and returns an encoding dictionary which can be used to encode other arrays
+ * @param array - original array before encoding
+ * @returns - dictionnary from string to number
+ */
+export function numericEncodingOfArray(array: (string | number)[][]) {
```
Comment (lpatiny) [inline src/utils/numericEncodingOfArray.ts:6]: In this case it is not an array but a matrix, please rename the variable

File: src/utils/numericEncodingOfArray.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+/**
+ * Numerically encodes the strings in the array and returns an encoding dictionary which can be used to encode other arrays
+ * @param array - original array before encoding
+ * @returns - dictionnary from string to number
+ */
```
Comment (lpatiny) [inline src/utils/numericEncodingOfArray.ts:5]: The method that have as first parameter a matrix should be named starting with matrix and be moved in the folder matrix.

File: src/utils/numericEncodingOfArray.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+  for (let i = 0; i < nRows; i++) {
+    for (let j = 0; j < nColumns; j++) {
+      if (typeof array[i][j] === 'string') {
+        if (array[i][j] in dictCategoricalToNumerical) {
+          array[i][j] = dictCategoricalToNumerical[array[i][j]];
+        } else {
+          dictCategoricalToNumerical[array[i][j]] = k;
+          array[i][j] = k;
+          k = k + 1;
```
Comment (lpatiny) [inline src/utils/numericEncodingOfArray.ts:21]:
```suggestion
          array[i][j] = k++;
```

**Review by lpatiny — CHANGES_REQUESTED** [review]
To keep the (discussable) naming approach of the package please use as names:
- `matrixNumericalEncode (matrix)`
- `matrixNumericalDecode (matrix, dictionary)`

In test cases please ensure that encode + decode yields exactly the all the same values.

File: src/matrix/matrixNumericalEncoding.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  let nRows = matrix.length;
+  let nColumns = matrix[0].length;
+
+  let dictCategoricalToNumerical: { [nameString: string]: number } = {};
+  let k = 0;
+
+  for (let i = 0; i < nRows; i++) {
+    for (let j = 0; j < nColumns; j++) {
+      if (typeof matrix[i][j] === 'number' && matrix[i][j] > k) {
+        k = Number(matrix[i][j]);
```
[the remark this answers is not in the document]
Comment (aiday-mar) [reply src/matrix/matrixNumericalEncoding.ts:16]: If I remove the number, the compile throws the following error : Type 'string | number' is not assignable to type 'number'.
  Type 'string' is not assignable to type 'number'.

[the remark this answers is not in the document]
Comment (targos) [reply src/matrix/matrixNumericalEncoding.ts:22]: You can use `as number` to make the compiler happy

**Pull request merged**